qtpy/__init__.py

```diff
@@ -256,7 +256,7 @@ def __init__(self, *, missing_package=None, **superclass_kwargs):
         QT6 = PYSIDE6 = True
 
     except ImportError:
-        API = 'pyqt5'
+        raise QtBindingsNotFoundError()
```

Comment:
It's not clear to me if `QT_API` is set at this point or not. If it is, then we should clear it here too.

Reply:
At this point `QT_API` is not set as an env variable yet (the `os.environ[QT_API]` call is only done if a successful binding import is done). However, `API` has as value `pyside6` at this point. Should we change the variable `API` value before raising the `QtBindingsNotFoundError` to something like `''` or `None` @ccordoba12 ?

Reply:
> At this point QT_API is not set as an env variable yet

Ok, that good to know.

>  Should we change the variable API value before raising the QtBindingsNotFoundError to something like '' or None

I think you could simply remove it from the environment by doing `os.environ.pop(`API`)`.

Reply:
`API` is just a variable which is initialized with `os.environ.get(QT_API, 'pyqt5')` and that is changed while the different bindings imports are being tried. If a binding import is successful, then the value of `API` is used to set the `QT_API` env var with something like `os.environ[QT_API] = API`. So asically there is no env var `API` set at any point

Reply:
Sorry, I thought `API` was also an environment variable. Then I think there's no need to change its value to something else because at this point the package will raise `QtBindingsNotFoundError` and `API` nor `QT_API` will be leaked to the environment (which was my concern).